### PR TITLE
Add: GDPR notice to contributing guide and a matching pull request template.

### PR DESCRIPTION
**Pull request by michicc**
Contents is adapted from the rsyslog/rsyslog project, which has mainly European contributors.

Whether such a statement is actually a good idea is up for debate. Not having one might open us to claims of privacy violations by collecting data without prior clear information. Having one might open us to claims that the statement is somehow wrong or non-conforming and as such a violation.

Which risk is the higher risk or if the GDPR even applies to use at all is still up in the air, especially as the whole legal situation is far from settled.

The full text in the pull request template might get tedious and deter from contributing alltogether. As such, a short template that just mentions privacy and license and otherwise refers to CONTRIBUTING.md is also an option.

### CODE REVIEW

**Review by LordAro — DISMISSED** [review]

File: CONTRIBUTING.md
Comment (TrueBrain) [inline CONTRIBUTING.md]: `Please note that the [EU GDPR](https://www.eugdpr.org/key-changes.html) "right to be forgotten" information cannot ..`
Possibly a better way of phrasing?

**Comment by TrueBrain** [comment]
I really like what you did @michicc , especially the rewrite from earlier drafts and no longer put GDPR in the center. I really like that we inform the user what we do with their information, and than refer to GDPR. Thank you for that.

The template part I am very unsure of. As you mention already, it is more bloat than it helps.
I would like to suggest to use a 'CLA' like system like in other repositories, where we ask any contributor to read CONTRIBUTING.md, and after they indicated they read it, allow their PR to be merged. This not only helps for this part, but also that they are fully aware they are committing to GPL. 
How it would look: if a user makes a PR, a bot will ask them to 'sign' a 'CLA'. I think it will be a simple website where they check a box to indicate they have read some simple bulletpoints (keep it simple, keep it short, much like the BaNaNaS ToS). After that, and any next time, the bot will automatically mark the PR as 'CLA approved', and that is done. Saves a lot of text in a PR, and has a higher chance of anyone reading it, I hope :D Best effort I guess. What do you think?

Either way, the first part for sure, I am very much in favor! But I would like to see more hands for this, as it kinda touches some ground we all try to stay away from. Sadly, we do have to touch it in this modern world :) (well, with how it is written now, I am not sad about it. I really like that we give users the chance to be more anonymous if they choose; no matter if GDPR is there or not!).

@frosch123 ? @andythenorth ? others? Any input is welcome.

File: CONTRIBUTING.md
Comment (andythenorth) [reply CONTRIBUTING.md]: +1 it's not a refusal to honour the right, it's arguing that there's a legitimate reasons for retention.  IANAL, but I am happy to help with GDPR stuff.

**Comment by michicc** [comment]
Reformulated "right to be forgotten". Legal basis is probably article 17, section 3(e).

The commit also includes an alternative, bare bones PR template.

File: CONTRIBUTING.md
Comment (nielsmh) [inline CONTRIBUTING.md]: Repeated phrase "we strongly recommend to use"

File: CONTRIBUTING.md
Comment (nielsmh) [inline CONTRIBUTING.md]: Rather than "the actual commit", perhaps "the actual change"?

Maybe also a sentence about commit histories longer than one, are they preserved when a change is merged, and whose responsibility is it to do any squashing?

File: CONTRIBUTING.md
Comment (planetmaker) [reply CONTRIBUTING.md]: It should read for proper English "This is a main feature *of* git"

**Comment by TrueBrain** [comment]
Looks good to me! I still don't think the template would be put in any use, so possibly hold off with that till we know how to shape that part of the landscape. I do still have to make sure anonymous at openttd dot org is semi-valid; I will do that the weekend of the 7 July.

**Comment by michicc** [comment]
Nobody likes my commit template 😢, as such here is a consolidated version without.

**Comment by planetmaker** [comment]
I guess this burocratic thing has to be added. Should we consider to leave-out the "refrain from contributing", thus the last sentence would read: "...please do commit anonymously to the OpenTTD project."?

I suggest to merge this only once the commit via anonymous@openttd.org works.

**Comment by TrueBrain** [comment]
The hot weather severely delayed me adding this. Scheduled for this weekend :)

**Review by TrueBrain — APPROVED** [review]
All done! Go Go Go :)

**Pull request merged**
